Docs: Adds test-development walkthrough documentation (#436)
fossas/spectrometer: docs/walkthroughs/test-dev-dependency.md

> # Working with Test and Development Dependency

Reviewer: [nit] Should be "dependencies".

> FOSSA CLI, by default, does not include the unused dependency in analysis. This is expected for scenarios in which, you have a dependency for development, testing, and other purposes - which ultimately are not used in your runtime. FOSSA CLI can infer dependency's inclusion for runtime or distribution by package manager's manifest file. When we are unable to infer if the dependency is only used in runtime, we choose the conservative approach and include them in our analysis reporting. 

Reviewer: [grammar] I don't think "dependency" can be used as a plural noun.

```suggestion
FOSSA CLI, by default, does not include unused dependencies in analysis. This is expected for scenarios in which, you have a dependency for development, testing, and other purposes - which ultimately are not used in your runtime. FOSSA CLI can infer dependency's inclusion for runtime or distribution by package manager's manifest file. When we are unable to infer if the dependency is only used in runtime, we choose the conservative approach and include them in our analysis reporting. 
```

CLI's language support documentation (in [strategy's section](./../references/strategies/README.md)), includes whether an analysis of a particular project will by default unused dependency.

## Example 

For the following `package.json` file:

```json
{
    "name": "example-project",
    "dependencies": {
        "pngjs": "6.0.0",
        "once": "1.4.0"
    },
    "devDependencies": {
        "mocha": "*"
    }
}
```

### When running `fossa analyze`:

We will report the dependency tree as follows:

```tree
.
 |-npm+once$1.4.0 (direct)

>  | \-npm+wrappy$1 (transitive)

Reviewer: Where is `wrappy` in the `package.json`? You may want to make the example file a `package-lock.json` file. I don't think the jump to where the transitive deps are coming from will be immediately obvious to most people reading this doc. This also goes for the dependency tree towards the bottom with dev deps in it. I was confused at first about where all of the other deps came from.

 \-npm+pngjs$6.0.0 (direct)
```

> Note that, we are not reporting "mocha" dependency or any other transitive dependency of mocha. This is intended behavior, as aforementioned, FOSSA CLI, by default only report dependency which is used in your runtime or distribution.

Reviewer: [nit] confusing sentence
```suggestion
Note that, we are not reporting the "mocha" dependency or any other transitive dependencies of mocha as it is a development dependency. By default, the FOSSA CLI only reports dependencies used in your runtime or distribution.
```

### How to include all dependencies?

> You can use the `--include-unused-deps` flag to explicitly include all _discovered_ dependencies found. This will include all unused dependencies (development, configuration, etc.)

Reviewer: What does discovered found? I think this would be better if you said something like the following.
```suggestion
You can use the `--include-unused-deps` flag to explicitly include all dependencies, regardless of how they are used. This will include all dependencies (development, configuration, etc.)
```

You can run analysis like this:
```bash
fossa analyze --include-unused-deps
```

With the flag, it will report the following dependency graph now (notice the inclusion of mocha and its dependencies)

```tree

> .

Author: Blocked on:
1. https://github.com/fossas/team-analysis/issues/811
2. https://github.com/fossas/team-analysis/issues/810
